### PR TITLE
[WC-2199]: Add show footer property to combobox-web

### PR DESCRIPTION
**Pull request by samuelreichert**
### Pull request type

New feature (non-breaking change which adds functionality)

---

### Description

Add a "Show footer" boolean property to combobox, when set to true, enables the developers to place widgets (like text, buttons, links) on the combobox menu footer)

Footer example:
Screenshot 2023-12-07 at 17 40 21

Structure mode:
Screenshot 2023-12-07 at 17 38 28

Design mode:
Screenshot 2023-12-07 at 17 38 05

### CODE REVIEW

File: packages/pluggableWidgets/combobox-web/src/components/SingleSelection/SingleSelection.tsx [hunk 1]
```diff
@@ -27,7 +30,7 @@ export function SingleSelection({
     return (
         <Fragment>
             <ComboboxWrapper
-                isOpen={isOpen}
+                isOpen={preview && showFooter ? true : isOpen}
```
Comment (gjulivan) [inline packages/pluggableWidgets/combobox-web/src/components/SingleSelection/SingleSelection.tsx:33]: isOpen || (preview && showFooter)

File: packages/pluggableWidgets/combobox-web/CHANGELOG.md [hunk 2]
```diff
@@ -5,3 +5,7 @@ All notable changes to this widget will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+
+### Added
+
+-   We added a new property named "Show footer" which enables the widget to show a footer inside combobox menu. With show footer set to true the developers can add custom footer content like text, button, links, etc.
```
Comment (r0b1n) [inline packages/pluggableWidgets/combobox-web/CHANGELOG.md:11]: We added a new property called **Show footer**, allowing the widget to display a footer inside the combobox menu. When **Show footer** is set to **Yes**, developers can include custom footer content such as text, buttons, links, etc.

File: packages/pluggableWidgets/combobox-web/src/components/SingleSelection/SingleSelectionMenu.tsx [hunk 3]
```diff
@@ -18,21 +21,31 @@ export function SingleSelectionMenu({
...
     getItemProps,
-    noOptionsText
+    noOptionsText,
+    preview,
+    showFooter,
+    showFooterContent
 }: ComboboxMenuProps): ReactElement {
-    const items = selector.options.getAll();
+    let items = selector.options.getAll();
+    if (preview && showFooter) {
```
Comment (gjulivan) [inline packages/pluggableWidgets/combobox-web/src/components/SingleSelection/SingleSelectionMenu.tsx:30]: you don't need this here. override the items on `AssociationPreviewSelector`

File: packages/pluggableWidgets/combobox-web/src/Combobox.xml [hunk 4]
```diff
@@ -111,6 +111,14 @@
                     <caption>Custom content</caption>
                     <description />
                 </property>
+                <property key="showFooter" type="boolean" defaultValue="false">
+                    <caption>Show footer</caption>
+                    <description />
+                </property>
+                <property key="showFooterContent" type="widgets" required="false">
```
Comment (r0b1n) [inline packages/pluggableWidgets/combobox-web/src/Combobox.xml:118]: Why `show` keyword in this name, this reads like it should be a boolean (show footer content: yes or no?). 
I believe `footerContent` or `menuFooterContent` more reflects what this property does.

File: packages/pluggableWidgets/combobox-web/src/components/ComboboxMenuWrapper.tsx [hunk 5]
```diff
@@ -23,6 +26,7 @@ export function ComboboxMenuWrapper(props: ComboboxMenuWrapperProps): ReactEleme
             <ul className="widget-combobox-menu-list" {...getMenuProps?.({}, { suppressRefError: true })}>
                 {isOpen ? isEmpty ? <NoOptionsPlaceholder>{noOptionsText}</NoOptionsPlaceholder> : children : null}
             </ul>
+            {showFooter && showFooterContent && <div className="widget-combobox-menu-footer">{showFooterContent}</div>}
```
Comment (r0b1n) [inline packages/pluggableWidgets/combobox-web/src/components/ComboboxMenuWrapper.tsx:29]: Same concerns about the naming of the prop.

File: packages/pluggableWidgets/combobox-web/src/Combobox.xml (hunk 4 above)
Comment (samuelreichert) [reply packages/pluggableWidgets/combobox-web/src/Combobox.xml:118]: i'll go with menuFooterContent

**Review by gjulivan — APPROVED** [review]

**Review by gjulivan — APPROVED** [review]

**Review by leonardomendix — APPROVED** [review]

**Pull request merged**